addons/website/models/website.py

```diff
@@ -315,25 +295,29 @@ def page_search_dependencies(self, cr, uid, view_id=False, context=None):
 
         return dep
 
-    def page_for_name(self, cr, uid, ids, name, module='website', context=None):
+    @api.multi
+    def page_for_name(self, name, module='website'):
         # whatever
         return '%s.%s' % (module, slugify(name, max_length=50))
 
-    def page_exists(self, cr, uid, ids, name, module='website', context=None):
+    @api.multi
+    def page_exists(self, name, module='website'):
         try:
             name = (name or "").replace("/page/website.", "").replace("/page/", "")
             if not name:
                 return False
-            return self.pool["ir.model.data"].get_object_reference(cr, uid, module, name)
+            return self.env["ir.model.data"].xmlid_to_res_model_res_id(module+'.'+name)
         except:
             return False
 
-    @openerp.tools.ormcache('id')
-    def _get_languages(self, cr, uid, id, context=None):
-        website = self.browse(cr, uid, id)
-        return [(lg.code, lg.name) for lg in website.language_ids]
+    @api.multi
+    @tools.ormcache('self.id')
```

Comment:
yes it's working fine. it's update lang of cache on change of website id. Thanks